homework/index.js

```diff
@@ -1,19 +1,21 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
```

Comment:
Hi Alusine, even though this is a pattern which you might encounter as you dive deep to programming where it is sometimes useful to wrap callbacks into promises but this homework doesn't require you to do so. Instead, you had to use the native browser `fetch` method which kind of does the same thing implicitly as your are doing here. Can you please try to use the same? More information on the `fetch` method can be found here - https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

Please be mindful of this statement which you can also find in the documentation - `An accurate check for a successful fetch() would include checking that the promise resolved, then checking that the Response.ok property has a value of true. `

This means, even if the promise has resolved, it is not sure if the response was successful. You will have to do another check of `response.ok` to make sure that you have got the repositories and contributors data.

Reply:
Hi Yash,
Thanks for taking the time to review my homework. I am currently loaded with other homework and exercises. I will soon complete the required tasks you mentioned and will let you know.

Reply:
Thank you. I realise that you must be stuffed with other home-works.